### PR TITLE
Feature/forgot password

### PR DESCRIPTION
**Pull request by ivanmrsulja**
## Description
Implemented "I forgot my password" functionality.

## How should this be tested?
There are unit and integration test for everything, but you can test using postman by firstly setting up a mail client, then sending a request to http://localhost:8081/api/user/forgot-password, and after that, sending a request to http://localhost:8081/api/user/reset-password together with the token provided in the email.
## Checklist

-   [x] I have tested these changes locally
-   [x] I have added tests that cover these changes (if applicable)
-   [x] I have updated the documentation (if applicable)
-   [x] I have added or updated the necessary comments in the code

### CODE REVIEW

**Review by chenejac — APPROVED** [review]
@ivanmrsulja thanks. Looks nice. My only comment is about naming submitForgottenPassword method and ForgotPasswordSubmissionDTO class, it looks strange to me.

**Comment by ivanmrsulja** [comment]
> @ivanmrsulja thanks. Looks nice. My only comment is about naming submitForgottenPassword method and ForgotPasswordSubmissionDTO class, it looks strange to me.

Maybe rename it to `initiatePasswordResetProcess` and `PasswordResetRequestDTO`? What do you think?

**Review by chenejac — APPROVED** [review]
@ivanmrsulja great. I will merge this after merging of https://github.com/chenejac/TeslaRIS-backend/pull/66

**Pull request merged**
